src/class-handler.php

```diff
@@ -0,0 +1,143 @@
+class Handler {
+
+	/**
+	 * Post_Duplicator object
+	 *
+	 * @var Post_Duplicator
+	 */
+	private $post_duplicator;
+
+	/**
+	 * Initializes the main class.
+	 *
+	 * @param Post_Duplicator $post_duplicator The Post_Duplicator object.
+	 */
+	public function __construct( Post_Duplicator $post_duplicator ) {
+		$this->post_duplicator = $post_duplicator;
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	private function register_hooks() {
+		\add_action( 'admin_action_duplicate_post_copy_for_rewrite', [ $this, 'rewrite_link_action_handler' ] );
+		\add_action( 'admin_init', [ $this, 'add_bulk_handlers' ] );
+	}
+
+	/**
+	 * Handles the action for copying a post for the Rewrite and Republish feature.
+	 *
+	 * @return void
+	 */
+	public function rewrite_link_action_handler() {
+		if ( ! \duplicate_post_is_current_user_allowed_to_copy() ) {
+			\wp_die( \esc_html__( 'Current user is not allowed to copy posts.', 'duplicate-post' ) );
+		}
+
+		if ( ! ( isset( $_GET['post'] ) || isset( $_POST['post'] ) || // Input var okay.
+			( isset( $_REQUEST['action'] ) && 'duplicate_post_copy_for_rewrite' === $_REQUEST['action'] ) ) ) { // Input var okay.
+			\wp_die( \esc_html__( 'No post to duplicate has been supplied!', 'duplicate-post' ) );
+		}
+
+		$id = ( isset( $_GET['post'] ) ? intval( \wp_unslash( $_GET['post'] ) ) : intval( \wp_unslash( $_POST['post'] ) ) ); // Input var okay.
+
+		\check_admin_referer( 'duplicate-post_rewrite_' . $id ); // Input var okay.
+
+		$post = \get_post( $id );
+
+		if ( ! $post ) {
+			\wp_die(
+				\esc_html(
+					__( 'Copy creation failed, could not find original:', 'duplicate-post' ) . ' '
+					. $id
+				)
+			);
+		}
+
+		if ( $post->post_status !== 'publish' ) {
+			\wp_die(
+				\esc_html(
+					__( 'You cannot create a copy for Rewrite and Republish if the original is not published.', 'duplicate-post' )
+				)
+			);
+		}
+
+		$new_id = $this->post_duplicator->create_duplicate_for_rewrite_and_republish( $post );
+
+		if ( \is_wp_error( $new_id ) ) {
+			\wp_die(
+				\esc_html(
+					__( 'Copy creation failed, could not create a copy.', 'duplicate-post' )
+				)
+			);
+		}
+
+		\wp_safe_redirect(
+			\add_query_arg(
+				array(
+					'rewriting' => 1,
+					'ids'       => $post->ID,
+				),
+				\admin_url( 'post.php?action=edit&post=' . $new_id . ( isset( $_GET['classic-editor'] ) ? '&classic-editor' : '' ) )
+			)
+		);
+		exit();
+	}
+
+	/**
+	 * Hooks the handler for the Rewrite and Republish action for all the selected post types.
+	 *
+	 * @return void
+	 */
+	public function add_bulk_handlers() {
+		$duplicate_post_types_enabled = Utils::get_enabled_post_types();
+
+		foreach ( $duplicate_post_types_enabled as $duplicate_post_type_enabled ) {
+			add_filter( "handle_bulk_actions-edit-{$duplicate_post_type_enabled}", [ $this, 'bulk_action_handler' ], 10, 3 );
+		}
+	}
+
+	/**
+	 * Bulk action handler for the Rewrite and Republish feature.
```

Comment:
I think the Yoast preference is to use a third-person verb for method descriptions e.g. "Adds... Shows... Filters... ecc.)